test/test_quantized_nn_mods.py

```diff
@@ -114,6 +118,16 @@ def test_linear_api(self, batch_size, in_features, out_features, use_bias):
         # Test JIT
         self.checkScriptable(qlinear, zip([X], [Z_ref]), check_save_load=True)
 
+        # Test from_float
+        float_linear = torch.nn.Linear(in_features, out_features).float()
```

Comment:
I don't think `.float()` here is necessary -- it's the default

Reply:
NOPE. This is an absolutely dangerous assumption. See the following:

```
import torch

print(torch.Tensor(3).dtype)

torch.set_default_tensor_type('torch.DoubleTensor')

print(torch.Tensor(3).dtype)

```

which outputs

```
torch.float32
torch.float64
```

And indeed, under *all tests* the default scalar type is in fact Double:

https://github.com/pytorch/pytorch/blob/0619b57c4c64be625050b32a6be6ee064b5ba2c3/test/common_utils.py#L45